src/OpenApi/src/Services/OpenApiDocumentProvider.cs

```diff
@@ -44,7 +44,8 @@ public async Task GenerateAsync(string documentName, TextWriter writer, OpenApiS
         // document to a file. See https://github.com/microsoft/OpenAPI.NET/issues/421 for
         // more info.
         var targetDocumentService = serviceProvider.GetRequiredKeyedService<OpenApiDocumentService>(documentName);
-        var document = await targetDocumentService.GetOpenApiDocumentAsync();
+        var scopedService = serviceProvider.CreateScope();
```

Comment:
When does this scope end?  This seems like the sort of thing that would need to be disposed.

Reply:
Ah good catch -- there should be a `using` at the beginning of this line. The scope terminates after the call to `GetOpenApiDocumentAsync` to mimic what happens at at runtime when an HTTP request is sent to retrieve the OpenAPI document.

This codepath is a bit unique because the `IDocumentProvider` is queried once per document at build-time when the app is launched via the `HostFactoryResolver` so there isn't an HTTP request-based scope at all (in fact, there's [no server running](https://github.com/dotnet/aspnetcore/blob/01da7cd4614bf573776ab3455239e05a8f2b0e9b/src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs#L67-L76)).